pkg/provider/cloud/gke/provider.go

```diff
@@ -482,6 +482,59 @@ func createClient(ctx context.Context, serviceAccount string, scope string) (*ht
 	return client, projectID, nil
 }
 
+func ListGKESizes(ctx context.Context, sa, zone string) (apiv1.GCPMachineSizeList, error) {
+	sizes := apiv1.GCPMachineSizeList{}
+
+	computeService, project, err := gcp.ConnectToComputeService(ctx, sa)
+	if err != nil {
+		return sizes, err
+	}
+
+	req := computeService.MachineTypes.List(project, zone)
+	err = req.Pages(ctx, func(page *compute.MachineTypeList) error {
+		for _, machineType := range page.Items {
+			mt := apiv1.GCPMachineSize{
+				Name:        machineType.Name,
+				Description: machineType.Description,
+				Memory:      machineType.MemoryMb,
+				VCPUs:       machineType.GuestCpus,
+			}
+			sizes = append(sizes, mt)
+		}
+		return nil
+	})
+
+	return sizes, err
+}
+
+func ListGKEDiskTypes(ctx context.Context, sa string, zone string) (apiv2.GKEDiskTypeList, error) {
+	diskTypes := apiv2.GKEDiskTypeList{}
+	// TODO: There are some issues at the moment with local-ssd and pd-extreme, that's why it is disabled at the moment.
```

Comment:
What are those issues? Would be helpful for whoever wants to work on this TODO in 3, 6, 9 months time, so they can know if the issues are resolved already.